ambry-utils/src/test/java/com.github.ambry.utils/ByteBufferInputStreamTest.java

```diff
@@ -61,6 +61,23 @@ public void byteBufferStreamTest()
       Assert.assertEquals(output[i], buf[i + 1001]);
     }
     Assert.assertEquals(stream3.read(), -1);
+
+    ByteBuffer byteBuf = ByteBuffer.wrap(buf);
+    ByteBufferInputStream stream4 = new ByteBufferInputStream(byteBuf.duplicate());
+    // ByteBuffer class overrides equal() to do content comparison.
+    Assert.assertEquals("The returned byte buffer must have the same content as the one initialized with", byteBuf,
+        stream4.getByteBuffer());
+    byteBuf.rewind();
+    ByteBufferInputStream stream5 = new ByteBufferInputStream(byteBuf.duplicate());
+    ByteBufferInputStream stream6 = new ByteBufferInputStream(stream5, 1024);
```

Comment:
Was just going with the pattern in the rest of this class. Also, I think the hardcoded value makes sense here as that is the source of truth. Using `byteBuf.remaining()` might mask issues or catch them later (like if `byteBuf` was not rewound).